clients/fides-js/src/lib/consent-utils.ts

```diff
@@ -242,17 +234,19 @@ export const hasSavedTcfPreferences = (experience: PrivacyExperience) =>
     hasCurrentPreference(experience[key])
   );
 
-export const hasActionNeededTcfPreferences = (experience: PrivacyExperience) =>
-  EXPERIENCE_KEYS_WITH_PREFERENCES.some((key) =>
-    hasActionNeededTcfPreference(experience[key])
-  );
-
 /**
  * Returns true if there are notices in the experience that require a user preference
+ * or if an experience's version hash does not match up.
  */
-export const hasActionNeededNotices = (experience: PrivacyExperience) => {
-  if (experience.component === ComponentType.TCF_OVERLAY) {
-    return hasActionNeededTcfPreferences(experience);
+export const resurfaceConsent = (
+  experience: PrivacyExperience,
+  cookie: FidesCookie
+) => {
+  if (
+    experience.component === ComponentType.TCF_OVERLAY &&
+    experience.meta?.version_hash
+  ) {
+    return experience.meta.version_hash !== cookie.tcf_version_hash;
```

Comment:
this is the crux of the change

Reply:
on the BE, have all experiences been migrated to have a `version_hash`? Does this mean that for users that have a Fides consent cookie on their browser, when they navigate to a site that has Fides installed, they will get the banner popup again?

Reply:
good point! this is only for TCF—the logic for notice banners is still the same as before. the backend doesn't have version hashes for notice based experiences yet. So I think we are safe here (since no one is using TCF yet)

Reply:
cool this makes sense, just making sure it's something we've thought about 👍